core/tee/tee_svc_cryp.c

```diff
@@ -1017,15 +1017,14 @@ TEE_Result tee_obj_attr_to_binary(struct tee_obj *o, void *data,
 		const struct tee_cryp_obj_type_attrs *ta = tp->type_attrs + n;
 		void *attr = (uint8_t *)o->attr + ta->raw_offs;
 
-		attr_ops[ta->ops_index].to_binary(attr, data, *data_len, &offs);
+		attr_ops[ta->ops_index].to_binary(attr, data, len, &offs);
 	}
 
-	if (offs > *data_len)
-		res = TEE_ERROR_SHORT_BUFFER;
-	else
-		res = TEE_SUCCESS;
+	if (data && offs > *data_len)
+		return TEE_ERROR_SHORT_BUFFER;
+
 	*data_len = offs;
```

Comment:
Often functions with an API like this updates the required length also when returning "short buffer". We don't take advantage of that right now, but it could avoid some future surprises. What I'm talking about is something like this:
```
*data_len = offs;
if (data && offs > len)
        return TEE_ERROR_SHORT_BUFFER;
return TEE_SUCCESS;
```